draft-ietf-masque-h3-datagram.md

```diff
@@ -76,46 +78,48 @@ document are to be interpreted as described in BCP 14 {{!RFC2119}} {{!RFC8174}}
 when, and only when, they appear in all capitals, as shown here.
 
 
-# Flow Identifiers {#flow-id}
+# Flows
 
-Flow identifiers represent bidirectional flows of datagrams within a single QUIC
-connection. These are conceptually similar to streams in the sense that they
-allow multiplexing of application data.  Flows lack any of the ordering
-or reliability guarantees of streams.
+Flows are bidirectional exchanges of datagrams within a single QUIC connection.
+These are conceptually similar to streams in the sense that they allow
+multiplexing of application data. Flows are identified within a connection by a
+numeric value, referred to as the flow ID.  A flow ID is a 62-bit integer (0
+to 2^62-1).
 
-Beyond this, a sender SHOULD ensure that DATAGRAM frames within a single flow
-are transmitted in order relative to one another. If multiple DATAGRAM frames
-can be packed into a single QUIC packet, the sender SHOULD group them by flow
-identifier to promote fate-sharing within a specific flow and improve the
-ability to process batches of datagram messages efficiently on the receiver.
+Flows lack any of the ordering or reliability guarantees of streams. Beyond
+this, a sender SHOULD ensure that DATAGRAM frames within a single flow are
+transmitted in order relative to one another. If multiple DATAGRAM frames can be
+packed into a single QUIC packet, the sender SHOULD group them by flow to
+promote fate-sharing within a specific flow and improve the ability to process
+batches of datagram messages efficiently on the receiver.
 
 
-# Flow Identifier Allocation {#flow-id-alloc}
+# Flow Allocation {#flow-id-alloc}
```

Comment:
Technically, you don't allocate a flow, you allocate a flow ID - which is part of what you need to do when creating a new flow. So while I like the concept of a flow in general, I think the allocation service should deal in identifiers. What do you think?

Reply:
I think this is probably the most controversial part, so I'm glad we can agree it needs discussion.

From an implementation perspective I do allocate for flows but that is an application-context dependent property (CONNECT-UDP would setup a flow<>UDP socket mapping for instance). But you're right that this text is specifically about IDs and I think that ignoring application details in this document is probably fine.

Let me sleep on things and circle back.

Reply:
I've changed this to `Flow ID allocation service` throughout